File: engine/memory.py
class MemorySystem:
    def __init__(self):
        self.trustmap = {}
        self.regret_lattice = []

    def update_trust(self, key, value):
        try:
            self.trustmap[key] = self.trustmap.get(key, 0) + value
            print(f"- Trustmap Updated: {key} +{value}")
        except Exception as e:
            print("⚠️ Error updating trust:", e)

    def log_regret(self, key, reason):
        try:
            self.regret_lattice.append((key, reason))
            print(f"- Regret Lattice: {key} → {reason}")
        except Exception as e:
            print("⚠️ Error logging regret:", e)

File: engine/memory.ts
// File: engine/memory.ts

export class MemorySystem {
  trustmap: Record<string, number> = {};
  regret_lattice: [string, string][] = [];

  trust(key: string, amount = 1) {
    if (!this.trustmap[key]) this.trustmap[key] = 0;
    this.trustmap[key] += amount;
  }

  regret(key: string, reason: string) {
    this.regret_lattice.push([key, reason]);
  }

  decay(factor = 0.95) {
    for (const key in this.trustmap) {
      this.trustmap[key] *= factor;
      if (this.trustmap[key] < 0.1) delete this.trustmap[key];
    }
  }
}
